### PR TITLE
Purescript 0.9.3 compatibility

### PR DESCRIPTION
**Pull request by adamw**
This is based on the fork by @sardonicpresence (https://github.com/sardonicpresence/purescript-sequences). There are three minor changes:
- tests fixed
- updated dependencies
- typos fixed

I know there's another effort under way (PR #23) which, as far as I understand, aims to eliminate `unsafePartial` with proper `Partial` constraints propagation. However it seems that the work there stalled, and it would be great to have a working implementation of sequences for psc 0.9.x meanwhile :)

To sum up:
- **the good**: tests pass (at least on my machine)
- **the bad**: `unsafePartial` is used

### CODE REVIEW

**Comment by hdgarrood** [comment]
Hi. Is this further along than #23 in any way? My impression was that #23 is pretty much there except that I want Partial constraints to accurately reflect partiality of functions; last time I checked, the tests in #23 did work, I thought.

Also, I'm sorry, but I would rather not merge this until we have Partial constraints in the right places.

**Comment by adamw** [comment]
Ah, I thought that they fail as the travis build was failing, but now I see that it was a built timeout, so maybe they are passing. I wanted to check this locally as well, but the base branch seems to be deleted, so I'm not even sure how to check out these changes - is there any way to download the patch maybe? Can't find it in the UI ...

**Comment by hdgarrood** [comment]
Maybe add ".patch" to the URL?

**Comment by adamw** [comment]
Ah yes that worked :) Even better, I found that you can just checkout the branch using git, even if the original fork is deleted:

# git remote add upstream git@github.com:hdgarrood/purescript-sequences.git
# git fetch upstream pull/23/head:LOCAL_NAME

**Comment by hdgarrood** [comment]
Oh cool, good to know!

**Comment by adamw** [comment]
Ok, checked #23, currently it doesn't even compile due to some `Partial` constraints not being propagated. I'll try to understand what happens there or just use my fork for now, time allowing :)

**Comment by hdgarrood** [comment]
Ok cool, thanks for checking :)

**Comment by Podlas29** [comment]
Hi. Is there any reason why this pr is not being merged? Anyone?
Thanks in advance and have a nice day. :)

**Comment by hdgarrood** [comment]
Yes, as I said earlier in the thread, it's because neither of the PRs I have received have Partial constraints in the right places. If anyone wants to pick this up I'd happily review, otherwise I will probably get around to doing it myself eventually.

**Comment by hdgarrood** [comment]
Fixed in #26.
